### ISSUE TITLE
False positive `reportUninitializedInstanceVariable` for `NamedTuple`s on v1.1.368

### ISSUE THREAD

**Issue by AlexWaygood** [microsoft/pyright#8184]
**Describe the bug**
Pyright's `reportUninitializedInstanceVariable` diagnostic emits new false positives for `NamedTuple` classes on v1.1.368. These false positive errors were not seen on v1.1.367.

**Code or Screenshots**

```python
# pyright: reportUninitializedInstanceVariable=true
from typing import NamedTuple

class Bar(NamedTuple):
    x: int
```

Pyright's complaint on the last line is:

```
Instance variable "x" is not initialized in the class body or __init__ method  (reportUninitializedInstanceVariable)
```

Playground link: https://pyright-play.net/?pyrightVersion=1.1.368&code=MQAgDgngTglg5gCwC4C4RQKZgPZSQVQDsZikYBDAGxgC8MATASUIGclzCBjDANXNnIAjShgC8SKAFcMAKABmUbAFsQSCGBJwQMJTjwgAcuSUMAKpLAiZMzpXIsWIAEL8AFEZP1zljAEoUMiBBIAAeaCRIMkA

But NamedTuple classes have synthesized `__init__` methods like dataclasses, so the instance variable `x` is guaranteed to be initialized on instances of the `Bar` class.

This bug is blocking typeshed from upgrading to the latest pyright version (see the CI failures in https://github.com/python/typeshed/pull/12171).

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.369.
